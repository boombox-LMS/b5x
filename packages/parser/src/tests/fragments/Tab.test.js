const {
  exampleTabMarkup,
} = require("../../../dist/componentBuilders/fragments/Tab");
const { RawFragmentSchema } = require("@b5x/types");

describe("The Tab fragment builds as expected", () => {
  const fragment = global.buildFragmentFromMarkupString(exampleTabMarkup);
  const rawFragment = fragment.packageForApi();

  test("The parsed tab fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed tab fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });
});
