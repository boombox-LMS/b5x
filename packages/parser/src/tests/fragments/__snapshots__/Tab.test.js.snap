// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Tab fragment builds as expected The parsed tab fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [
    "test-fragment-uri",
  ],
  "contentType": "Tab",
  "contents": "",
  "data": Object {
    "title": "Tab 1 title",
  },
  "dependencyUris": Array [],
  "digest": "b927b11ce15323c54f79dd25d0aa3300",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": false,
  "isStateful": false,
  "uri": "test-fragment-uri",
}
`;
